stress: failed test in cockroach/gossip/gossip.test: TestGossipStorageCleanup

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/f6cb86c1aa5a39cae56a972c6043837601912a75

Stress build found a failed test:

```
=== RUN   TestGossipStorageCleanup
I160818 05:17:39.919009 gossip/simulation/network.go:68  simulating gossip network with 3 nodes
W160818 05:17:39.919554 gossip/gossip.go:1035  not connected to cluster; use --join to specify a connected node
W160818 05:17:39.919776 gossip/gossip.go:1035  not connected to cluster; use --join to specify a connected node
W160818 05:17:39.919938 gossip/gossip.go:1035  not connected to cluster; use --join to specify a connected node
I160818 05:17:39.920457 gossip/client.go:74  node 1: starting client to 127.0.0.1:44658
I160818 05:17:39.920639 gossip/client.go:74  node 2: starting client to 127.0.0.1:44658
I160818 05:17:39.920750 gossip/client.go:74  node 3: starting client to 127.0.0.1:44658
W160818 05:17:39.921724 gossip/gossip.go:831  invalid bootstrap address: &{typ:tcp addr:10.0.0.1000:3333333}, invalid port 3333333
I160818 05:17:39.921778 gossip/client.go:74  node 1: starting client to 127.0.0.1:33381
W160818 05:17:39.921823 gossip/gossip.go:831  invalid bootstrap address: &{typ:tcp addr:10.0.0.1000:3333333}, invalid port 3333333
I160818 05:17:39.921863 gossip/client.go:74  node 2: starting client to 127.0.0.1:42855
W160818 05:17:39.921897 gossip/gossip.go:831  invalid bootstrap address: &{typ:tcp addr:10.0.0.1000:3333333}, invalid port 3333333
I160818 05:17:39.921954 gossip/client.go:74  node 3: starting client to 127.0.0.1:38962
I160818 05:17:39.923086 gossip/client.go:98  node 3: closing client to node 3 (127.0.0.1:44658): stopping outgoing client to node 3 (127.0.0.1:44658); loopback connection
I160818 05:17:39.923870 gossip/client.go:98  node 1: closing client to node 3 (127.0.0.1:44658): stopping outgoing client to node 3 (127.0.0.1:44658); already have incoming
I160818 05:17:39.924063 gossip/gossip.go:1003  node has connected to cluster via gossip
I160818 05:17:39.924490 gossip/gossip.go:1003  node has connected to cluster via gossip
I160818 05:17:39.924870 gossip/gossip.go:1003  node has connected to cluster via gossip
I160818 05:17:39.925531 gossip/simulation/network.go:193  gossip network simulation: total infos sent=96, received=0
I160818 05:17:39.958482 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.958522 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.958541 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.958574 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.961358 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 7 (127.0.0.1:39769)
I160818 05:17:39.961898 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 7 (127.0.0.1:39769)
I160818 05:17:39.962220 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.962261 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.962301 gossip/client.go:74  node 3: starting client to 127.0.0.1:39769
I160818 05:17:39.962336 gossip/client.go:74  node 3: starting client to 127.0.0.1:39769
I160818 05:17:39.963947 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.963983 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.969078 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.969142 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.978884 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 7 (127.0.0.1:39769); already have active connection, skipping
I160818 05:17:39.979590 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.979630 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.979654 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.979680 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.979813 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.979858 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.980630 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.980670 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.980690 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.980721 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.983346 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.983386 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.983407 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.983446 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.984097 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.984135 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.990772 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.990807 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:39.993812 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:39.993852 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:40.001413 gossip/gossip.go:975  node 2: starting client to distant node 6 to tighten network graph
I160818 05:17:40.001482 gossip/client.go:74  node 2: starting client to 127.0.0.1:42891
I160818 05:17:40.006674 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 7 (127.0.0.1:39769); already have active connection, skipping
I160818 05:17:40.006719 gossip/gossip.go:975  node 1: starting client to distant node 6 to tighten network graph
I160818 05:17:40.006753 gossip/client.go:74  node 1: starting client to 127.0.0.1:42891
I160818 05:17:40.006949 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 7 (127.0.0.1:39769); already have active connection, skipping
I160818 05:17:40.007054 gossip/gossip.go:975  node 1: starting client to distant node 6 to tighten network graph
I160818 05:17:40.007081 gossip/client.go:74  node 1: starting client to 127.0.0.1:42891
I160818 05:17:40.007389 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 7 (127.0.0.1:39769); already have active connection, skipping
I160818 05:17:40.007806 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 5 (127.0.0.1:39767)
I160818 05:17:40.008404 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 2 (127.0.0.1:44150); already have active connection, skipping
I160818 05:17:40.008757 gossip/gossip.go:975  node 2: starting client to distant node 6 to tighten network graph
I160818 05:17:40.008791 gossip/client.go:74  node 2: starting client to 127.0.0.1:42891
I160818 05:17:40.009033 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 5 (127.0.0.1:39767)
I160818 05:17:40.009120 gossip/client.go:74  node 3: starting client to 127.0.0.1:39767
I160818 05:17:40.009153 gossip/client.go:74  node 3: starting client to 127.0.0.1:39767
I160818 05:17:40.009424 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 7 (127.0.0.1:39769); already have active connection, skipping
I160818 05:17:40.010928 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:40.010968 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:40.010985 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:40.011030 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:40.011837 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:40.011889 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:40.011912 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:40.011940 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:40.013260 gossip/client.go:100  node 3: closing client to 127.0.0.1:39767: rpc error: code = 2 desc = duplicate connection from node at {tcp 127.0.0.1:42855}
I160818 05:17:40.013290 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 7 (127.0.0.1:39769); already have active connection, skipping
I160818 05:17:40.013373 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 7 (127.0.0.1:39769); already have active connection, skipping
I160818 05:17:40.013447 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 7 (127.0.0.1:39769); already have active connection, skipping
I160818 05:17:40.013498 gossip/client.go:100  node 1: closing client to 127.0.0.1:42891: rpc error: code = 2 desc = duplicate connection from node at {tcp 127.0.0.1:38962}
I160818 05:17:40.013552 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 5 (127.0.0.1:39767); already have active connection, skipping
I160818 05:17:40.013592 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 2 (127.0.0.1:44150); already have active connection, skipping
I160818 05:17:40.013635 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 5 (127.0.0.1:39767); already have active connection, skipping
I160818 05:17:40.013676 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 5 (127.0.0.1:39767); already have active connection, skipping
I160818 05:17:40.014098 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 7 (127.0.0.1:39769); already have active connection, skipping
I160818 05:17:40.015952 gossip/client.go:98  node 3: closing client to node 10 (127.0.0.1:33383): received forward from node 10 to 2 (127.0.0.1:44150); already have active connection, skipping
I160818 05:17:40.027917 http2_client.go:1022  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I160818 05:17:40.027931 http2_client.go:1022  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I160818 05:17:40.028088 http2_client.go:1022  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I160818 05:17:40.028134 http2_client.go:1022  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I160818 05:17:40.028174 http2_client.go:1022  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I160818 05:17:40.028353 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:42891: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:42891" <nil>}
I160818 05:17:40.028379 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39767: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39767" <nil>}
I160818 05:17:40.028395 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:44658: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:44658" <nil>}
I160818 05:17:40.028410 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39769: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39769" <nil>}
I160818 05:17:40.028421 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:33383: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:33383" <nil>}
I160818 05:17:40.028554 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:40.028596 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:40.028696 gossip/client.go:100  node 3: closing client to 127.0.0.1:33383: rpc error: code = 14 desc = grpc: RPC failed fast due to transport failure
I160818 05:17:40.029082 gossip/gossip.go:975  node 3: starting client to distant node 10 to tighten network graph
I160818 05:17:40.029120 gossip/client.go:74  node 3: starting client to 127.0.0.1:33383
I160818 05:17:40.029180 gossip/client.go:100  node 3: closing client to 127.0.0.1:33383: rpc error: code = 14 desc = grpc: RPC failed fast due to transport failure
I160818 05:17:41.028394 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:42891: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:42891" <nil>}
I160818 05:17:41.028413 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39769: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39769" <nil>}
I160818 05:17:41.028428 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:33383: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:33383" <nil>}
I160818 05:17:41.028453 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:44658: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:44658" <nil>}
I160818 05:17:41.028477 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39767: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39767" <nil>}
I160818 05:17:42.363869 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:42891: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:42891" <nil>}
I160818 05:17:42.514561 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:33383: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:33383" <nil>}
I160818 05:17:42.720986 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39769: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39769" <nil>}
I160818 05:17:42.737074 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39767: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39767" <nil>}
I160818 05:17:42.769936 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:44658: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:44658" <nil>}
I160818 05:17:44.941125 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:33383: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:33383" <nil>}
I160818 05:17:45.009849 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:44658: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:44658" <nil>}
I160818 05:17:45.013340 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39769: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39769" <nil>}
I160818 05:17:45.049695 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:42891: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:42891" <nil>}
I160818 05:17:45.333300 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39767: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39767" <nil>}
I160818 05:17:48.498613 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39769: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39769" <nil>}
I160818 05:17:48.609773 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:33383: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:33383" <nil>}
I160818 05:17:48.787528 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:42891: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:42891" <nil>}
I160818 05:17:49.282460 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39767: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39767" <nil>}
I160818 05:17:49.315855 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:44658: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:44658" <nil>}
I160818 05:17:54.881743 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39769: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39769" <nil>}
I160818 05:17:55.472705 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:42891: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:42891" <nil>}
I160818 05:17:55.491364 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:33383: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:33383" <nil>}
I160818 05:17:56.160429 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39767: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39767" <nil>}
I160818 05:17:56.470253 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:44658: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:44658" <nil>}
I160818 05:18:03.863548 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:42891: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:42891" <nil>}
I160818 05:18:06.227333 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39767: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39767" <nil>}
I160818 05:18:06.753890 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39769: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39769" <nil>}
I160818 05:18:06.947251 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:44658: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:44658" <nil>}
I160818 05:18:06.967449 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:33383: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:33383" <nil>}
I160818 05:18:20.374890 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39769: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39769" <nil>}
I160818 05:18:20.381947 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:44658: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:44658" <nil>}
I160818 05:18:20.408335 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:33383: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:33383" <nil>}
I160818 05:18:21.338587 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:42891: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:42891" <nil>}
I160818 05:18:22.397597 /go/src/google.golang.org/grpc/clientconn.go:645  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:39767: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:39767" <nil>}
I160818 05:18:25.015900 /go/src/google.golang.org/grpc/clientconn.go:745  grpc: addrConn.transportMonitor exits due to: context canceled
I160818 05:18:25.015945 /go/src/google.golang.org/grpc/clientconn.go:745  grpc: addrConn.transportMonitor exits due to: context canceled
W160818 05:18:25.015966 gossip/infostore.go:291  node unavailable; try another peer
W160818 05:18:25.015995 gossip/infostore.go:291  node unavailable; try another peer
I160818 05:18:25.016066 /go/src/google.golang.org/grpc/clientconn.go:745  grpc: addrConn.transportMonitor exits due to: context canceled
I160818 05:18:25.016125 /go/src/google.golang.org/grpc/clientconn.go:745  grpc: addrConn.transportMonitor exits due to: context canceled
W160818 05:18:25.016134 gossip/infostore.go:291  node unavailable; try another peer
I160818 05:18:25.016164 /go/src/google.golang.org/grpc/clientconn.go:745  grpc: addrConn.transportMonitor exits due to: context canceled
--- FAIL: TestGossipStorageCleanup (45.10s)
	testing.go:117: gossip/storage_test.go:271, condition failed to evaluate within 45s: expected 2, got 9 (info: []util.UnresolvedAddr{util.UnresolvedAddr{NetworkField:"tcp", AddressField:"127.0.0.1:33381"}, util.UnresolvedAddr{NetworkField:"tcp", AddressField:"127.0.0.1:44658"}, util.UnresolvedAddr{NetworkField:"tcp", AddressField:"127.0.0.1:42891"}, util.UnresolvedAddr{NetworkField:"tcp", AddressField:"127.0.0.1:39809"}, util.UnresolvedAddr{NetworkField:"tcp", AddressField:"127.0.0.1:35971"}, util.UnresolvedAddr{NetworkField:"tcp", AddressField:"127.0.0.1:33138"}, util.UnresolvedAddr{NetworkField:"tcp", AddressField:"127.0.0.1:33383"}, util.UnresolvedAddr{NetworkField:"tcp", AddressField:"127.0.0.1:39767"}, util.UnresolvedAddr{NetworkField:"tcp", AddressField:"127.0.0.1:39769"}})
FAIL


ERROR: exit status 1
```

Run Details:

```
94 runs so far, 0 failures, over 5s
197 runs so far, 0 failures, over 10s
297 runs so far, 0 failures, over 15s
400 runs so far, 0 failures, over 20s
505 runs so far, 0 failures, over 25s
613 runs so far, 0 failures, over 30s
714 runs so far, 0 failures, over 35s
817 runs so far, 0 failures, over 40s
920 runs so far, 0 failures, over 45s
1018 runs so far, 0 failures, over 50s
1088 runs so far, 0 failures, over 55s
1154 runs so far, 0 failures, over 1m0s
1231 runs so far, 0 failures, over 1m5s
1307 runs so far, 0 failures, over 1m10s
1382 runs so far, 0 failures, over 1m15s
1445 runs so far, 0 failures, over 1m20s
1504 runs so far, 0 failures, over 1m25s
1562 runs so far, 0 failures, over 1m30s
1610 runs completed, 1 failures, over 1m34s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/gossip/storage_test.go
+++ b/gossip/storage_test.go
@@ -17,7 +17,6 @@
 package gossip_test
 
 import (
-	"net"
 	"reflect"
 	"sort"
 	"strings"
@@ -219,14 +218,7 @@ func TestGossipStorageCleanup(t *testing.T) {
 	network := simulation.NewNetwork(numNodes, false)
 	defer network.Stop()
 
-	// Create a definitely closed address.
-	ln, err := net.Listen("tcp", util.TestAddr.String())
-	if err != nil {
-		t.Fatal(err)
-	}
-	closedAddr := ln.Addr()
-	ln.Close()
-
+	const notReachableAddr = "localhost:0"
 	const invalidAddr = "10.0.0.1000:3333333"
 	// Set storage for each of the nodes.
 	addresses := make(unresolvedAddrSlice, len(network.Nodes))
@@ -237,7 +229,7 @@ func TestGossipStorageCleanup(t *testing.T) {
 		if err := stores[i].WriteBootstrapInfo(&gossip.BootstrapInfo{
 			Addresses: []util.UnresolvedAddr{
 				util.MakeUnresolvedAddr("tcp", network.Nodes[(i+1)%numNodes].Addr().String()), // node i+1 address
-				util.MakeUnresolvedAddr(closedAddr.Network(), closedAddr.String()),            // valid but closed address
+				util.MakeUnresolvedAddr("tcp", notReachableAddr),                              // unreachable address
 				util.MakeUnresolvedAddr("tcp", invalidAddr),                                   // invalid address
 			},
 		}); err != nil {

